evennia/contrib/fieldfill.py

```diff
@@ -0,0 +1,561 @@
+        new_fieldname = None
+        new_fieldvalue = None
+        # Get field name
+        new_fieldname = "|w" + field["fieldname"] + ":|n"
+        if len(field["fieldname"]) + 5 > field_name_width:
+            field_name_width = len(field["fieldname"]) + 5
+        # Get field value
+        if formdata[field["fieldname"]] != None:
+            new_fieldvalue = str(formdata[field["fieldname"]])
+        # Use blank message if field is blank and once is present
+        if new_fieldvalue == None and "blankmsg" in field:
+            new_fieldvalue = "|x" + str(field["blankmsg"]) + "|n"
+        elif new_fieldvalue == None:
+            new_fieldvalue = " "
+        # Add name and value to table
+        formtable.add_row(new_fieldname, new_fieldvalue)
+        
+    formtable.reformat_column(0, align="r", width=field_name_width)
+    # formtable.reformat_column(1, pad_left=0)
+        
+    return pretext + "|/" + str(formtable) + "|/" + posttext
+
+"""
+EXAMPLE FUNCTIONS / COMMAND STARTS HERE
+"""
+    
+def verify_online_player(caller, value):
+    """
+    Example 'verify function' that matches player input to an online character
+    or else rejects their input as invalid.
+    
+    Args:
+        caller (obj): Player entering the form data
+        value (str): String player entered into the form, to be verified
+    
+    Returns:
+        matched_character (obj or False): dbref to a currently logged in
+            character object - reference to the object will be stored in
+            the form instead of a string. Returns False if no match is
+            made.
+    """
+    # Get a list of sessions
+    session_list = SESSIONS.get_sessions()
+    char_list = []
+    matched_character = None
+    
+    # Get a list of online characters
+    for session in session_list:
+        if not session.logged_in:
+            # Skip over logged out characters
+            continue
+        # Append to our list of online characters otherwise
+        char_list.append(session.get_puppet())
+
+    # Match player input to a character name
+    for character in char_list:
+        if value.lower() == character.key.lower():
+            matched_character = character
+    
+    # If input didn't match to a character
+    if not matched_character:
+        # Send the player an error message unique to this function
+        caller.msg("No character matching '%s' is online." % value)
+        # Returning False indicates the new value is not valid
+        return False
+    
+    # Returning anything besides True or False will replace the player's input with the returned value
+    # In this case, the value becomes a reference to the character object
+    # You can store data besides strings and integers in the 'formdata' dictionary this way!
+    return matched_character
+
+# Form template for the example 'delayed message' form
+SAMPLE_FORM = [
+{"fieldname":"Character", "fieldtype":"text", "max":30, "blankmsg":"(Name of an online player)",
```

Comment:
Just an idea, but for this type of form it may be a little easier to read by formatting it a little: 
```
{"fieldname": "Character", 
 "fieldtype": "text", 
 "max":30, "blankmsg": "(Name of an online player)",
 "required": True, 
 "verifyfunc": verify_online_player
},
{"fieldname": "Delay", 
 "fieldtype": "number", 
 "min": 3, 
 "max": 30, 
 "default": 10, 
 "cantclear": True
},
{"fieldname": "Message", 
 "fieldtype":"text", 
 "min":3, 
 "max":200, 
 "blankmsg":"(Message up to 200 characters)"
}]```

Reply:
Oh, that looks much nicer!